StorageUtility/Windows/Device.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+<License>
+Copyright 2015 Virtium Technology
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+</License>
+*/
+#include "Device.h"
+#include "DeviceDataConverter.h"
+
+namespace vtStor
+{
+
+cDevice::cDevice(PSP_DEVINFO_DATA PDevInfoData, PSP_DEVICE_INTERFACE_DATA PDevInterfaceData, PSP_INTERFACE_DEVICE_DETAIL_DATA PDevDetailData)
+{
+    m_Data.clear();
+    m_Data.insert({ eDeviceDataType::PSP_DEVINFO_DATA, cDeviceDataConverter<PSP_DEVINFO_DATA>::ToVoidPointer(PDevInfoData) });
+    m_Data.insert({ eDeviceDataType::PSP_DEVICE_INTERFACE_DATA, cDeviceDataConverter<PSP_DEVICE_INTERFACE_DATA>::ToVoidPointer(PDevInterfaceData) });
+    m_Data.insert({ eDeviceDataType::PSP_INTERFACE_DEVICE_DETAIL_DATA, cDeviceDataConverter<PSP_INTERFACE_DEVICE_DETAIL_DATA>::ToVoidPointer(PDevDetailData) });
```

Comment:
1. I still believe this will cause a run time bug.
2. I don't understand the purpose of using the cDeviceDataConverter class.